docs/reference/alert-schema.asciidoc

```diff
@@ -0,0 +1,118 @@
+|{ecs-ref}/ecs-url.html[`url.*`]|`url.*`|ECS `url.*` fields copied from the source document when available.
+|{ecs-ref}/ecs-user.html[`user.*`]|`user.*`| ECS `user.*` fields copied from the source document when available.
+|{ecs-ref}/ecs-user_agent.html[`user_agent.*`]|`user_agent.*`| ECS `user_agent.*` fields copied from the source document when available.
+|{ecs-ref}/ecs-vulnerability.html[`vulnerability.*`]|`vulnerability.*`|ECS `vulnerability.*` fields copied from the source document when available.
+|`signal.ancestors.*`|`kibana.alert.ancestors.*`|
+|`signal.depth`| `kibana.alert.depth`|
+|`signal.original_event.*` | `kibana.alert.original_event.*`|
+|`signal.original_time`|`kibana.alert.original_time`| The value copied from the source event (`@timestamp`).
+|`signal.reason` |`kibana.alert.reason`|
+|`signal.rule.author` |`kibana.alert.rule.author`|
+|`signal.rule.building_block_type`|`kibana.alert.building_block_type`||`signal.rule.created_at`|`kibana.alert.rule.created_at`|
+|`signal.rule.created_by` |`kibana.alert.rule.created_by`|
+|`signal.rule.description` |`kibana.alert.rule.description`|
+|`signal.rule.enabled` |`kibana.alert.rule.enabled`|
+|`signal.rule.false_positives`| `kibana.alert.rule.false_positives`|
+|`signal.rule.from` |`kibana.alert.rule.from`|
+|`signal.rule.id` |`kibana.alert.rule.uuid` |
+|`signal.rule.immutable` |`kibana.alert.rule.immutable`|
+|`signal.rule.interval` |`kibana.alert.rule.interval`|
+|`signal.rule.license` |`kibana.alert.rule.license`|
+|`signal.rule.max_signals`|`kibana.alert.rule.max_signals`|
+|`signal.rule.name` |`kibana.alert.rule.name` |
+|`signal.rule.note` |`kibana.alert.rule.note`|
+|`signal.rule.references` |`kibana.alert.rule.references`|
+|`signal.rule.risk_score` |`kibana.alert.risk_score`|
+|`signal.rule.rule_id` |`kibana.alert.rule.rule_id`|
+|`signal.rule.rule_name_override`|`kibana.alert.rule.rule_name_override`|
+|`signal.rule.severity` |`kibana.alert.severity`| Alert severity, populated by the `rule_type` at alert creation. Must have a value of `low`, `medium`, `high`, `critical`.
+|`signal.rule.tags` |`kibana.alert.rule.tags`|
+|`signal.rule.threat.*` |`kibana.alert.rule.threat.*`|
+|`signal.rule.timeline_id` |`kibana.alert.rule.timeline_id`|
+|`signal.rule.timeline_title`|`kibana.alert.rule.timeline_title`|
+|`signal.rule.timestamp_override`|`kibana.alert.rule.timestamp_override`|
+|`signal.rule.to` |`kibana.alert.rule.to`|
+|`signal.rule.type` |`kibana.alert.rule.type`|
+|`signal.rule.updated_at` |`kibana.alert.rule.updated_at`|
+|`signal.rule.updated_by` |`kibana.alert.rule.updated_by`|
+|`signal.rule.version` |`kibana.alert.rule.version`|
+|`signal.status` |`kibana.alert.workflow_status`|
+|`signal.threshold_result.*`|`kibana.alert.threshold_result.*`|
+|`signal.group.id` |`kibana.alert.group.id`|
+|`signal.group.index` |`kibana.alert.group.index`|
+|`signal.rule.index` |`kibana.alert.rule.parameters.index`|
```

Comment:
@MikePaquette do you think we should document the fields within `kibana.alert.rule.parameters` publicly? These fields are copied directly from the rule as a record of the rule at the time of execution. Perhaps we could document `kibana.alert.rule.parameters` as a single field and say something like "Contains the rule parameters at the time the alert was generated" but not document individual fields within `parameters`.

Reply:
I like the idea, but some of these fields are not documented anywhere else.  For example building_block_type, to, from, timeline_id, timeline_title, max_signals, so I think it's best if this document serves as the place where these are defined for now.  So my preference is that we keep all of them here and add relevant definitions where they are not covered by the rule basic and advanced settings docs.